src/libopensc/card-idprime.c

```diff
@@ -646,12 +652,14 @@ static int idprime_init(sc_card_t *card)
 	_sc_card_add_rsa_alg(card, 1024, flags, 0);
 	_sc_card_add_rsa_alg(card, 2048, flags, 0);
 	if (card->type == SC_CARD_TYPE_IDPRIME_930
+	    || card->type == SC_CARD_TYPE_IDPRIME_930_2
 	    || card->type == SC_CARD_TYPE_IDPRIME_940) {
 		_sc_card_add_rsa_alg(card, 4096, flags, 0);
 	}
 
 	/* Set up algorithm info for EC */
-	if (card->type == SC_CARD_TYPE_IDPRIME_940) {
+	if (card->type == SC_CARD_TYPE_IDPRIME_930
```

Comment:
should this one be the `SC_CARD_TYPE_IDPRIME_930_2`?

Reply:
My bad, fixed